[FEATURE] Ajout de la vue de la page de création de profil cible sur Pix Admin (PIX-4543)

clarissedmn opened the pull request:
## :unicorn: Problème
Depuis pix admin, nous souhaitons créer une page de création de profil.

## :robot: Solution
Reprendre l'existant et l'ajouter aux nouvelles maquettes.

## :100: Pour tester
Aller sur /target-profiles/new et créer un profil cible.

Commit:
Use Pix-ui Fonts

Co-authored-by: Jérémie Jadé <jeremie.jade@pix.fr>

Jeremiejade committed:
Use new design to display creation target profile

co-authored-by: Guillaume Lazzara <guillaume.lazzara@pix.fr>
Co-authored-by: Clarisse Damon <clarisse.damon@pix.fr>

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr4182.review.pix.fr
- App (.org): https://app-pr4182.review.pix.org
- Orga: https://orga-pr4182.review.pix.fr
- Certif: https://certif-pr4182.review.pix.fr
- Admin: https://admin-pr4182.review.pix.fr
- API: https://api-pr4182.review.pix.fr/api/

 Please check it out!

Commit:
Introduce card component

Co-authored-by: Jérémie Jadé <jeremie.jade@pix.fr>
Co-authored-by: Guillaume Lazzara <guillaume.lazzara@pix.fr>

Commit:
Add spacings in first block

Co-authored-by: Jérémie Jadé <jeremie.jade@pix.fr>
Co-authored-by: Guillaume Lazzara <guillaume.lazzara@pix.fr>

lisequesnel commented on admin/app/styles/globals/fonts.scss:
```diff
@@ -1,6 +0,0 @@
-$roboto: 'Roboto', Arial, sans-serif;
-$open-sans: 'Open Sans', Arial, sans-serif;
-
-$font-semi-bold : 600;
-$font-medium: 500;
-$font-light: 300;
```
**question** : ces variables sont supprimées ici mais sont sûrement utilisées, est-ce qu'elle existent également dans le nouvel import ?

glazzara replied:
Oui, elles existent bien. Nous avions fait des recherches dans le code avant de supprimer :)

lisequesnel commented on admin/app/styles/components/card.scss:
```diff
@@ -0,0 +1,23 @@
+.card {
+  box-shadow: 0 4px 8px rgba(7, 20, 46, 0.08);
+  border-radius: 8px;
+  background: #FFFFFF;
```
**suggestion** : pour toutes les couleurs, utiliser les variables de couleur `$xxxxxx` définies dans pix ui et importées dans le fichier `palette.scss`

lisequesnel commented on admin/app/styles/components/card.scss:
```diff
@@ -0,0 +1,23 @@
+    color: #FFFFFF;
+    font-family: $font-open-sans;
+    font-size: 20px;
+    font-style: normal;
```
**question** : cette ligne est-elle utile ?

lisequesnel commented on admin/app/styles/components/card.scss:
```diff
@@ -0,0 +1,23 @@
+  &__title {
+    color: #FFFFFF;
+    font-family: $font-open-sans;
+    font-size: 20px;
```
**suggestion** : c'est mieux d'utiliser les `rem` au lieu des `px` pour les `font-size`, pour une question d'accessibilité (le zoom fonctionnera correctement de cette manière).

Pour la conversion, c'est 1rem = 16px

lisequesnel commented on admin/app/styles/components/card.scss:
```diff
@@ -0,0 +1,23 @@
+    font-style: normal;
+    font-weight: $font-semi-bold;
+    line-height: 28px;
+    letter-spacing: 0;
```
**question** : cette ligne est-elle utile ?

lisequesnel commented on admin/app/styles/components/target-profiles/create-target-profile-form.scss:
```diff
@@ -1,17 +1,82 @@
+    margin-bottom: 0;
+
+    &__label {
+      border: 1px solid #5E6C84;
```
**suggestion** : idem ici pour les couleurs

lisequesnel commented on admin/app/styles/components/target-profiles/create-target-profile-form.scss:
```diff
@@ -1,17 +1,82 @@
+  &__is-public-label {
+    font-family: $font-roboto;
+    font-weight: $font-normal;
+    font-size: 14px;
```
**suggestion** : idem ici pour les font-size en rem

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+        <PixSelect
+          class="form-control"
+          @id="targetProfileCategory"
+          @label={{"Catégorie :"}}
```
```suggestion
          @label="Catégorie :"
```

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+        class="form-field"
+        @id="targetProfileName"
+        @label="Nom (obligatoire) : "
+        required
```
```suggestion
        required={{true}}
        aria-required={{true}}
```

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+          @options={{this.optionsList}}
+          @emptyOptionLabel={{"-"}}
+          @emptyOptionNotSelectable={{true}}
+          required={{true}}
```
```suggestion
          required={{true}}
          aria-required={{true}}
```

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+          @id="organizationId"
+          @label="Identifiant de l'organisation de référence : "
+          @errorMessage=""
+          required
```
```suggestion
          required={{true}}
          aria-required={{true}}
```

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+        @label="Lien de l'image du profil cible : "
+        @information="L'url à saisir doit être celle d'OVH. Veuillez
+          vous rapprocher des équipes tech et produit pour la réalisation de celle-ci."
+        {{on "change" this.updateImageUrl}}
```
**question** : pourquoi parfois on a `@value={{@targetProfile.description}}` et parfois on passe par `{{on "change" this.updateImageUrl}}` ?

Anne-Gaelle-S replied:
Yes c'est exactement ça (on a écrit une ADR sur le changement des Inputs d'ember aux inputs natifs : https://github.com/1024pix/pix-ui/blob/dev/docs/adr/0001-utiliser-les-balises-html-natives.md ).
Le `@value` ne permet plus le double binding sur les PixInputs (c'est depuis la version  de Pix UI 11.1.0). 

Par contre j'ai appris récemment que le `value=` pouvait toujours servir pour mettre le champ vide au besoin (exemple dans `admin/app/components/organizations/team-actions-section.hbs` où on a un PixInput avec un `value={{@userEmailToAdd}}` qui est remis à null `this.userEmailToAdd = null;` dans `admin/app/controllers/authenticated/organizations/get/team.js`).

La personne qui en parlera le mieux c'est @yschoueri  au besoin :D

lisequesnel commented:
On a aucun retour visuel qui nous signale qu'on a bien sélectionné un fichier : 
<img width="460" alt="image" src="https://user-images.githubusercontent.com/23451175/157882445-b9d21587-3336-44a6-b3ca-ada749417a6a.png">

lisequesnel commented on admin/app/components/target-profiles/create-target-profile-form.hbs:
```diff
@@ -1,100 +1,102 @@
+      <div class="create-target-profile__inline-layout">
+        <div class="file-upload-group">
+          <label class="file-upload-group__label {{if @isFileInvalid "s-invalid"}}">Importer un profil cible .JSON
+            <Input class="file-upload-group__input" {{on "change" @onLoadFile}} @type="file" id="skillsList" />
```
**question** : Pourquoi on n'utilise pas `PixButtonUpload` ?

Commit:
Code review : css and aria-required

Co-authored-by: Jérémie Jadé <jeremie.jade@pix.fr>
#Co-authored-by: Khadija Bayssir <khadija.bayssir@pix.fr>
Co-authored-by: Clarisse Damon <clarisse.damon@pix.fr>
#Co-authored-by: François de Metz <francois.de-metz@pix.fr>
#Co-authored-by: Thomas Evano <thomas.evano@pix.fr>
#Co-authored-by: Nicolas Legrand <nicolas.legrand@pix.fr>
#Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

Commit:
Use PixButtonUpload

Co-authored-by: Jérémie Jadé <jeremie.jade@pix.fr>
#Co-authored-by: Khadija Bayssir <khadija.bayssir@pix.fr>
Co-authored-by: Clarisse Damon <clarisse.damon@pix.fr>
#Co-authored-by: François de Metz <francois.de-metz@pix.fr>
#Co-authored-by: Thomas Evano <thomas.evano@pix.fr>
#Co-authored-by: Nicolas Legrand <nicolas.legrand@pix.fr>
#Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

lisequesnel commented on admin/app/controllers/authenticated/target-profiles/new.js:
```diff
@@ -19,21 +20,26 @@ export default class NewController extends Controller {
+      const json = JSON.parse(event.target.result);
+      const skillIds = json.flatMap((tube) => tube.skills);
+      if (skillIds.length === 0) {
+        throw new Error('Le fichier de compétence est vide !');
```
**suggestion** : "Le fichier est vide", il me semble pas que ce fichier soit appelé "fichier de compétence" côté métier, je me dis que ça peut porter à confusion
